### ISSUE TITLE
Detect duplicated source reference comments in PO files

### ISSUE THREAD

**Issue by luisg123v** [OCA/pylint-odoo#384]
We already detect duplicated source terms (msgid), thanks to `duplicate-po-message-definition` on #374 , e.g.:

```
#. module: mymodule
#: model:ir.model.fields,field_description:mymodule.field_res_partner__salesperson_id
msgid "Internal Salesperson"
msgstr "Vendedor Interno"

#. module: mymodule
#: model:ir.model.fields,field_description:mymodule.field_res_users__salesperson_id
msgid "Internal Salesperson"
msgstr "Vendedor Interno"
```

However, when msgid's are different but comments are the same, we are not catching such cases:

```
#. module: mymodule
#: model:ir.model.fields,field_description:mymodule.field_res_partner__salesperson_id
msgid "Internal Salesperson"
msgstr "Vendedor Interno"

#. module: mymodule
#: model:ir.model.fields,field_description:mymodule.field_res_partner__salesperson_id
msgid "Internal Salesperson 2"
msgstr "Vendedor Interno 2"
```

 Even though the latter is syntactically correct regarding PO format, it raises the same error (see #289 for the full error message), because it means two terms are trying to translate the same record and field.

Note that this only applies for comments starting with:

`# : model:`

Other terms like the following ones don't need to be considered, because they don't refer to a specific record:
- `#: code:`: refers to translatable terms inside code, and code could have the same term several times. Besides, starting from v14, [lines are ignored within translations](https://github.com/odoo/odoo/pull/35582), so all terms are included as if they were on line zero.
- `#: model_terms:`: usually refers to terms inside views. Same case, the same view could have the same term in several places (e.g. page and group).

**Comment by luisg123v** [comment]
@moylop260 @edy1192 @deivislaya @antonag32
